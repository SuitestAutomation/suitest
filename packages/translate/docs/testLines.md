# Test Lines

Translating test line definitions into human readable language.

## Prerequisites

In order to translate a line, you'll need the following data:

* Line definition, e.g. loaded from feed `/results/{resultId}`.
* Application configuration, e.g. from feed `/test-pack-runs/{testPackRunId}`, see `effectiveAppConfig` property.
    This information is needed because any configuration variables used in test need to be translated.
* Optional, a list of elements used during test execution, so that actual element names could be displayed instead of
    just ids. This information will be available from the feed `/results/{resultId}` shortly.
* Optional, a list of snippets used during test execution, so that actual snippet names could be displayed instead of
    just ids.

Reference the feeds can be found in [our documentation].

## Translating test line definition to a plain text

For example:
```javascript
const {testLineToPlainText} = require('@suitest/translate');

const translateLine = testLineToPlainText({type: 'sleep', timeout: 2000}, appConfiguration);

console.log(translateLine); // Sleep for 2s
```

## Translating test line definition to a formatted text

```javascript
const {testLineToFormattedText} = require('@suitest/translate');

const translateLine = testLineToFormattedText({
    type: 'assert',
    then: 'success',
    condition: {
        subject: {type: 'application'},
        type: 'exited'
    },
}, appConfiguration);

console.log(translatedLine); // Assert: Application has exited
```

## Translating test line definition to a HTML

```javascript
const {testLineToHtml} = require('@suitest/translate');

const translateLine = testLineToHtml({
    type: 'wait',
    then: 'success',
    timeout: 2000,
    condition: {
        subject: {type: 'cookie', val: 'my-cookie'},
        type: '=',
        val: 'test'
    },
}, appConfiguration);

console.log(translatedLine);
```

Output (in HTML):
```html
<figure class="suitest-test-line">
	<figcaption>Assert: Cookie</figcaption>
	<figure class="suitest-test-line__dictionary">
		<dl>
			<dt>Timeout</dt>
			<dd><span class="suitest-test-line__text--bold">2s</span></dd>
		</dl>
	</figure>
	<figure class="suitest-test-line__table">
		<table>
			<tr>
				<td><span class="suitest-test-line__text--bold">my-cookie</span> cookie</td>
				<td>=</td>
				<td><span class="suitest-test-line__text--bold">test</span></td>
			</tr>
		</table>
	</figure>
</figure>
```

In order to apply basic styling, make sure to include `@suitest/translate/suitest-test-line.css` file from
the NPM package into your HTML page.

## Other formats

We are using our own flavour of [unist] AST to translate test lines into multiple formats. Thus, you can implement
your own renderers if provided by default do not fit your needs.

```javascript
const {testLineToAst} = require('@suitest/translate');

const suitestAst = testLineToAst({type: 'sleep', timeout: 2000});

console.log(suitestAst);
```

Would produce:
```json5
{
  type: 'test-line',
  title: [
    {type: 'text', value: 'Sleep for '},
    {type: 'bold', value: '2'},
    {type: 'text', value: 's'},
  ],
}
```

[our documentation]: https://suite.st/docs/suitest-network-api/api-reference/
[unist]: https://github.com/syntax-tree/unist
