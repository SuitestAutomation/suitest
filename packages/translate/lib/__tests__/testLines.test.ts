import {testLineToAst} from '../testLineToAst';
import {appConfig, elements, snippets, testLines} from './testLines';

describe('Test Lines translation', () => {
	for (const [name, line] of Object.entries(testLines)) {
		it(`Should translate test line "${name}"`, () => {
			expect(testLineToAst(line(), appConfig, elements, snippets)).toMatchSnapshot();
		});
	}
});
