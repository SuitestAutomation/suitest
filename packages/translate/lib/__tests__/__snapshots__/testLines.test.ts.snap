// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Lines translation Should translate test line "Assert ... then continue" 1`] = `
Object {
  "children": Array [],
  "title": Array [
    Object {
      "type": "text",
      "value": "Assert: Application has exited",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Assert ... then exit" 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "When passed",
                },
              ],
              "type": "cell",
            },
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "stop test",
                },
              ],
              "type": "cell",
            },
          ],
          "type": "row",
        },
      ],
      "label": undefined,
      "type": "dictionary",
    },
  ],
  "title": Array [
    Object {
      "type": "text",
      "value": "Assert: Application has exited",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Assert ... then fail" 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "When passed",
                },
              ],
              "type": "cell",
            },
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "fail",
                },
              ],
              "type": "cell",
            },
          ],
          "type": "row",
        },
      ],
      "label": undefined,
      "type": "dictionary",
    },
  ],
  "title": Array [
    Object {
      "type": "text",
      "value": "Assert: Application has exited",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Assert ... then warn" 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "When passed",
                },
              ],
              "type": "cell",
            },
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "warn",
                },
              ],
              "type": "cell",
            },
          ],
          "type": "row",
        },
      ],
      "label": undefined,
      "type": "dictionary",
    },
  ],
  "title": Array [
    Object {
      "type": "text",
      "value": "Assert: Application has exited",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Assert ... timeout ... then continue" 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Timeout",
                },
              ],
              "type": "cell",
            },
            Object {
              "children": Array [
                Object {
                  "type": "bold",
                  "value": "1s",
                },
              ],
              "type": "cell",
            },
          ],
          "type": "row",
        },
      ],
      "label": undefined,
      "type": "dictionary",
    },
  ],
  "title": Array [
    Object {
      "type": "text",
      "value": "Assert: Application has exited",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Browser command: Accept modal dialog only if ..." 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "title": Array [
        Object {
          "type": "text",
          "value": "Application has exited",
        },
      ],
      "type": "condition",
    },
  ],
  "title": Array [
    Object {
      "type": "text",
      "value": "Accept modal dialog",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Browser command: Accept modal dialog" 1`] = `
Object {
  "children": Array [],
  "title": Array [
    Object {
      "type": "text",
      "value": "Accept modal dialog",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Browser command: Accept prompt dialog with text ... only if ..." 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Message",
                },
              ],
              "type": "cell",
            },
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "prompt text",
                },
              ],
              "type": "cell",
            },
          ],
          "type": "row",
        },
      ],
      "label": undefined,
      "type": "dictionary",
    },
    Object {
      "children": Array [],
      "title": Array [
        Object {
          "type": "text",
          "value": "Application has exited",
        },
      ],
      "type": "condition",
    },
  ],
  "title": Array [
    Object {
      "type": "text",
      "value": "Accept modal dialog with message",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Browser command: Accept prompt dialog with text ..." 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Message",
                },
              ],
              "type": "cell",
            },
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "prompt text",
                },
              ],
              "type": "cell",
            },
          ],
          "type": "row",
        },
      ],
      "label": undefined,
      "type": "dictionary",
    },
  ],
  "title": Array [
    Object {
      "type": "text",
      "value": "Accept modal dialog with message",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Browser command: Dismiss modal dialog only if ..." 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "title": Array [
        Object {
          "type": "text",
          "value": "Application has exited",
        },
      ],
      "type": "condition",
    },
  ],
  "title": Array [
    Object {
      "type": "text",
      "value": "Dismiss modal dialog",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Browser command: Dismiss modal dialog" 1`] = `
Object {
  "children": Array [],
  "title": Array [
    Object {
      "type": "text",
      "value": "Dismiss modal dialog",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Browser command: Go back only if ..." 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "title": Array [
        Object {
          "type": "text",
          "value": "Application has exited",
        },
      ],
      "type": "condition",
    },
  ],
  "title": Array [
    Object {
      "type": "text",
      "value": "Go back",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Browser command: Go back" 1`] = `
Object {
  "children": Array [],
  "title": Array [
    Object {
      "type": "text",
      "value": "Go back",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Browser command: Go forward only if ..." 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "title": Array [
        Object {
          "type": "text",
          "value": "Application has exited",
        },
      ],
      "type": "condition",
    },
  ],
  "title": Array [
    Object {
      "type": "text",
      "value": "Go forward",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Browser command: Go forward" 1`] = `
Object {
  "children": Array [],
  "title": Array [
    Object {
      "type": "text",
      "value": "Go forward",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Browser command: Refresh only if ..." 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "title": Array [
        Object {
          "type": "text",
          "value": "Application has exited",
        },
      ],
      "type": "condition",
    },
  ],
  "title": Array [
    Object {
      "type": "text",
      "value": "Refresh",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Browser command: Refresh" 1`] = `
Object {
  "children": Array [],
  "title": Array [
    Object {
      "type": "text",
      "value": "Refresh",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Clear app data" 1`] = `
Object {
  "children": Array [],
  "title": Array [
    Object {
      "type": "text",
      "value": "Clear application data",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Click on element ... every ... exactly ..." 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Repeat",
                },
              ],
              "type": "cell",
            },
            Object {
              "children": Array [
                Object {
                  "type": "bold",
                  "value": "4",
                },
                Object {
                  "type": "text",
                  "value": "x",
                },
              ],
              "type": "cell",
            },
          ],
          "type": "row",
        },
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Every",
                },
              ],
              "type": "cell",
            },
            Object {
              "children": Array [
                Object {
                  "type": "bold",
                  "value": "1.5s",
                },
              ],
              "type": "cell",
            },
          ],
          "type": "row",
        },
      ],
      "label": Array [
        Object {
          "type": "text",
          "value": "Condition settings",
        },
      ],
      "type": "dictionary",
    },
  ],
  "title": Array [
    Object {
      "type": "text",
      "value": "Click on ",
    },
    Object {
      "type": "bold",
      "value": "element",
    },
    Object {
      "type": "text",
      "value": " exactly ",
    },
    Object {
      "type": "bold",
      "value": "4",
    },
    Object {
      "type": "text",
      "value": "x",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Click on element ... once" 1`] = `
Object {
  "children": Array [],
  "title": Array [
    Object {
      "type": "text",
      "value": "Click on ",
    },
    Object {
      "type": "bold",
      "value": "element",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Click on element ... only if ..." 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "title": Array [
        Object {
          "type": "text",
          "value": "Application has exited",
        },
      ],
      "type": "condition",
    },
  ],
  "title": Array [
    Object {
      "type": "text",
      "value": "Click on ",
    },
    Object {
      "type": "bold",
      "value": "element",
    },
    Object {
      "type": "text",
      "value": " only if condition is met",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Click on element ... until ... every ... max ..." 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Repeat",
                },
              ],
              "type": "cell",
            },
            Object {
              "children": Array [
                Object {
                  "type": "bold",
                  "value": "6",
                },
                Object {
                  "type": "text",
                  "value": "x",
                },
              ],
              "type": "cell",
            },
          ],
          "type": "row",
        },
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Every",
                },
              ],
              "type": "cell",
            },
            Object {
              "children": Array [
                Object {
                  "type": "bold",
                  "value": "5.7s",
                },
              ],
              "type": "cell",
            },
          ],
          "type": "row",
        },
      ],
      "label": Array [
        Object {
          "type": "text",
          "value": "Condition settings",
        },
      ],
      "type": "dictionary",
    },
    Object {
      "children": Array [],
      "title": Array [
        Object {
          "type": "text",
          "value": "Application has exited",
        },
      ],
      "type": "condition",
    },
  ],
  "title": Array [
    Object {
      "type": "text",
      "value": "Click on ",
    },
    Object {
      "type": "bold",
      "value": "element",
    },
    Object {
      "type": "text",
      "value": " until condition is met",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Click on position ... once" 1`] = `
Object {
  "children": Array [],
  "title": Array [
    Object {
      "type": "text",
      "value": "Click on ",
    },
    Object {
      "type": "bold",
      "value": "position",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Comment" 1`] = `
Object {
  "children": Array [],
  "title": Array [
    Object {
      "type": "text",
      "value": "This is a comment",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Execute command ..." 1`] = `
Object {
  "children": Array [
    Object {
      "label": Array [
        Object {
          "type": "text",
          "value": "JavaScript expression",
        },
      ],
      "type": "code-block",
      "value": "someJS();",
    },
  ],
  "title": Array [
    Object {
      "type": "text",
      "value": "Execute JavaScript expression",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Move to element ... only if ..." 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "title": Array [
        Object {
          "type": "text",
          "value": "Application has exited",
        },
      ],
      "type": "condition",
    },
  ],
  "title": Array [
    Object {
      "type": "text",
      "value": "Move pointer to ",
    },
    Object {
      "type": "bold",
      "value": "element",
    },
    Object {
      "type": "text",
      "value": " only if condition is met",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Move to element ..." 1`] = `
Object {
  "children": Array [],
  "title": Array [
    Object {
      "type": "text",
      "value": "Move pointer to ",
    },
    Object {
      "type": "bold",
      "value": "element",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Move to position ..." 1`] = `
Object {
  "children": Array [],
  "title": Array [
    Object {
      "type": "text",
      "value": "Move pointer to ",
    },
    Object {
      "type": "bold",
      "value": "position",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Open URL ..." 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Exact URL",
                },
              ],
              "type": "cell",
            },
            Object {
              "children": Array [
                Object {
                  "type": "bold",
                  "value": "https://suite.st",
                },
              ],
              "type": "cell",
            },
          ],
          "type": "row",
        },
      ],
      "label": undefined,
      "type": "dictionary",
    },
  ],
  "title": Array [
    Object {
      "type": "text",
      "value": "Open URL",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Open app at homepage" 1`] = `
Object {
  "children": Array [],
  "title": Array [
    Object {
      "type": "text",
      "value": "Open application",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Open app at relative URL ..." 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Relative path",
                },
              ],
              "type": "cell",
            },
            Object {
              "children": Array [
                Object {
                  "type": "bold",
                  "value": "/some/path",
                },
              ],
              "type": "cell",
            },
          ],
          "type": "row",
        },
      ],
      "label": undefined,
      "type": "dictionary",
    },
  ],
  "title": Array [
    Object {
      "type": "text",
      "value": "Open application at relative URL",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Poll URL ... until response is ..." 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Poll URL",
                },
              ],
              "type": "cell",
            },
            Object {
              "children": Array [
                Object {
                  "type": "bold",
                  "value": "https://suite.st",
                },
              ],
              "type": "cell",
            },
          ],
          "type": "row",
        },
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Until receive",
                },
              ],
              "type": "cell",
            },
            Object {
              "children": Array [
                Object {
                  "type": "bold",
                  "value": "expected response",
                },
              ],
              "type": "cell",
            },
          ],
          "type": "row",
        },
      ],
      "label": undefined,
      "type": "dictionary",
    },
  ],
  "title": Array [
    Object {
      "type": "text",
      "value": "Poll URL every 0.5s",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Run snippet ... every ... exactly ..." 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Repeat",
                },
              ],
              "type": "cell",
            },
            Object {
              "children": Array [
                Object {
                  "type": "bold",
                  "value": "2",
                },
                Object {
                  "type": "text",
                  "value": "x",
                },
              ],
              "type": "cell",
            },
          ],
          "type": "row",
        },
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Every",
                },
              ],
              "type": "cell",
            },
            Object {
              "children": Array [
                Object {
                  "type": "bold",
                  "value": "3s",
                },
              ],
              "type": "cell",
            },
          ],
          "type": "row",
        },
      ],
      "label": Array [
        Object {
          "type": "text",
          "value": "Condition settings",
        },
      ],
      "type": "dictionary",
    },
  ],
  "title": Array [
    Object {
      "type": "text",
      "value": "Run test ",
    },
    Object {
      "type": "bold",
      "value": "Snippet 1",
    },
    Object {
      "type": "text",
      "value": " exactly ",
    },
    Object {
      "type": "bold",
      "value": "2",
    },
    Object {
      "type": "text",
      "value": "x",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Run snippet ... once" 1`] = `
Object {
  "children": Array [],
  "title": Array [
    Object {
      "type": "text",
      "value": "Run test ",
    },
    Object {
      "type": "bold",
      "value": "Snippet 1",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Run snippet ... only if ..." 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "title": Array [
        Object {
          "type": "text",
          "value": "Application has exited",
        },
      ],
      "type": "condition",
    },
  ],
  "title": Array [
    Object {
      "type": "text",
      "value": "Run test ",
    },
    Object {
      "type": "bold",
      "value": "Snippet 1",
    },
    Object {
      "type": "text",
      "value": " only if condition is met",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Run snippet ... until ... every ... max ..." 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Repeat",
                },
              ],
              "type": "cell",
            },
            Object {
              "children": Array [
                Object {
                  "type": "bold",
                  "value": "2",
                },
                Object {
                  "type": "text",
                  "value": "x",
                },
              ],
              "type": "cell",
            },
          ],
          "type": "row",
        },
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Every",
                },
              ],
              "type": "cell",
            },
            Object {
              "children": Array [
                Object {
                  "type": "bold",
                  "value": "3s",
                },
              ],
              "type": "cell",
            },
          ],
          "type": "row",
        },
      ],
      "label": Array [
        Object {
          "type": "text",
          "value": "Condition settings",
        },
      ],
      "type": "dictionary",
    },
    Object {
      "children": Array [],
      "title": Array [
        Object {
          "type": "text",
          "value": "Application has exited",
        },
      ],
      "type": "condition",
    },
  ],
  "title": Array [
    Object {
      "type": "text",
      "value": "Run test ",
    },
    Object {
      "type": "bold",
      "value": "Snippet 1",
    },
    Object {
      "type": "text",
      "value": " until condition is met",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Send text ... to element ... once" 1`] = `
Object {
  "children": Array [],
  "title": Array [
    Object {
      "type": "text",
      "value": "Send text ",
    },
    Object {
      "type": "bold",
      "value": "text to send",
    },
    Object {
      "type": "text",
      "value": " to ",
    },
    Object {
      "type": "bold",
      "value": "element",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Send text ... to window every ... exactly ..." 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Repeat",
                },
              ],
              "type": "cell",
            },
            Object {
              "children": Array [
                Object {
                  "type": "bold",
                  "value": "2",
                },
                Object {
                  "type": "text",
                  "value": "x",
                },
              ],
              "type": "cell",
            },
          ],
          "type": "row",
        },
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Every",
                },
              ],
              "type": "cell",
            },
            Object {
              "children": Array [
                Object {
                  "type": "bold",
                  "value": "1s",
                },
              ],
              "type": "cell",
            },
          ],
          "type": "row",
        },
      ],
      "label": Array [
        Object {
          "type": "text",
          "value": "Condition settings",
        },
      ],
      "type": "dictionary",
    },
  ],
  "title": Array [
    Object {
      "type": "text",
      "value": "Send text ",
    },
    Object {
      "type": "bold",
      "value": "text to send",
    },
    Object {
      "type": "text",
      "value": " to ",
    },
    Object {
      "type": "bold",
      "value": "window",
    },
    Object {
      "type": "text",
      "value": " exactly ",
    },
    Object {
      "type": "bold",
      "value": "2",
    },
    Object {
      "type": "text",
      "value": "x",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Send text ... to window once" 1`] = `
Object {
  "children": Array [],
  "title": Array [
    Object {
      "type": "text",
      "value": "Send text ",
    },
    Object {
      "type": "bold",
      "value": "text to send",
    },
    Object {
      "type": "text",
      "value": " to ",
    },
    Object {
      "type": "bold",
      "value": "window",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Send text ... to window only if ..." 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "title": Array [
        Object {
          "type": "text",
          "value": "Application has exited",
        },
      ],
      "type": "condition",
    },
  ],
  "title": Array [
    Object {
      "type": "text",
      "value": "Send text ",
    },
    Object {
      "type": "bold",
      "value": "text to send",
    },
    Object {
      "type": "text",
      "value": " to ",
    },
    Object {
      "type": "bold",
      "value": "window",
    },
    Object {
      "type": "text",
      "value": " only if condition is met",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Send text ... to window until ... every ... max ..." 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Repeat",
                },
              ],
              "type": "cell",
            },
            Object {
              "children": Array [
                Object {
                  "type": "bold",
                  "value": "4",
                },
                Object {
                  "type": "text",
                  "value": "x",
                },
              ],
              "type": "cell",
            },
          ],
          "type": "row",
        },
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Every",
                },
              ],
              "type": "cell",
            },
            Object {
              "children": Array [
                Object {
                  "type": "bold",
                  "value": "1.2s",
                },
              ],
              "type": "cell",
            },
          ],
          "type": "row",
        },
      ],
      "label": Array [
        Object {
          "type": "text",
          "value": "Condition settings",
        },
      ],
      "type": "dictionary",
    },
    Object {
      "children": Array [],
      "title": Array [
        Object {
          "type": "text",
          "value": "Application has exited",
        },
      ],
      "type": "condition",
    },
  ],
  "title": Array [
    Object {
      "type": "text",
      "value": "Send text ",
    },
    Object {
      "type": "bold",
      "value": "text to send",
    },
    Object {
      "type": "text",
      "value": " to ",
    },
    Object {
      "type": "bold",
      "value": "window",
    },
    Object {
      "type": "text",
      "value": " until condition is met",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Set text ... to element ... only if ..." 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "title": Array [
        Object {
          "type": "text",
          "value": "Application has exited",
        },
      ],
      "type": "condition",
    },
  ],
  "title": Array [
    Object {
      "type": "text",
      "value": "Set text ",
    },
    Object {
      "type": "bold",
      "value": "text to send",
    },
    Object {
      "type": "text",
      "value": " to ",
    },
    Object {
      "type": "bold",
      "value": "element",
    },
    Object {
      "type": "text",
      "value": " only if condition is met",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Set text ... to element ..." 1`] = `
Object {
  "children": Array [],
  "title": Array [
    Object {
      "type": "text",
      "value": "Set text ",
    },
    Object {
      "type": "bold",
      "value": "text to send",
    },
    Object {
      "type": "text",
      "value": " to ",
    },
    Object {
      "type": "bold",
      "value": "element",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Sleep ..." 1`] = `
Object {
  "children": Array [],
  "title": Array [
    Object {
      "type": "text",
      "value": "Sleep ",
    },
    Object {
      "type": "bold",
      "value": "2s",
    },
  ],
  "type": "test-line",
}
`;

exports[`Test Lines translation Should translate test line "Wait until ... max ... then continue" 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Timeout",
                },
              ],
              "type": "cell",
            },
            Object {
              "children": Array [
                Object {
                  "type": "bold",
                  "value": "1s",
                },
              ],
              "type": "cell",
            },
          ],
          "type": "row",
        },
      ],
      "label": undefined,
      "type": "dictionary",
    },
  ],
  "title": Array [
    Object {
      "type": "text",
      "value": "Assert: Application has exited",
    },
  ],
  "type": "test-line",
}
`;
